org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java

```diff
@@ -327,6 +330,15 @@ private boolean getInlineProposal(IInvocationContext context, ASTNode node, Coll
 							CheckConditionsOperation check = new CheckConditionsOperation(refactoring, CheckConditionsOperation.FINAL_CONDITIONS);
 							final CreateChangeOperation create = new CreateChangeOperation(check, RefactoringStatus.FATAL);
 							create.run(new NullProgressMonitor());
+							int referenceCount = 0;
+							Change change = create.getChange();
+							Change[] refactoringChanges = ((DynamicValidationRefactoringChange)change).getChildren();
+							for (Change refactoringChange : refactoringChanges) {
+								referenceCount += ((CompilationUnitChange)refactoringChange).getChangeGroups().length;
+							}
+							if (referenceCount <= 1 && refactoring.isDeclarationSelected()) {
```

Comment:
@CsCherrYY Below is a corner case. The change group will be 2 (1 for remove imports, 1 for remove the declaration) even though no references are using the constant field. The "Inline" menu won't be hidden. 
```java
import java.util.HashMap;
import java.util.Map;

public class Hello {
    public static final Map map = new HashMap<>();
}
```

Reply:
@testforstephen Thanks a lot for kind reminding, I would like to implement this clearer and avoid corner case like this by changing some APIs in jdt later.